evaluator/builtin_string.go

```diff
@@ -684,11 +681,7 @@ func builtinChar(args []types.Datum, ctx context.Context) (d types.Datum, err er
 		}
 	}
 
-	resultBytes, err = codec.ConvertInt64ToBytes(intSlice, binary.BigEndian, 4)
-	if err != nil {
-		return d, errors.Trace(err)
-	}
-	resultStr = hack.String(resultBytes)
+	resultStr = hack.String(convertInt64ToBytes(intSlice))
```

Comment:
I'm afraid `hack.String` is not fit for here, it may cause  dangling reference.
`hack.String` is commonly used during function call, **avoid malloc a temporarily string**, for example, 

```
b := []byte{..}
func f(data string) {...}
f(hack.String(b))
```

1. []byte and string are actually pointer, reference to some data 
2. that data won't be GC during call f(), **because `b` still refer to it**.

but return a hack.String() value is not the case:

```
func f() {
    b := []byte{...}
    return hack.String(b)
}
s := f()
```
s reference to the underlying data of b, but GC may recycle the data **because b no longer valid**